grapher/dataTable/DataTable.jsdom.test.tsx

```diff
@@ -82,62 +112,101 @@ describe(DataTable, () => {
             const cell = view.find("tbody .dimension-delta").first()
-            expect(cell.text()).toBe("-9.18 pp")
+            expect(cell.text()).toBe("-8.00 pp")
         })
 
         it("renders relative change values", () => {
             const cell = view.find("tbody .dimension-deltaRatio").first()
-            expect(cell.text()).toBe("-62%")
+            expect(cell.text()).toBe("-27%")
         })
     })
 
     describe("when the table doesn't have data for all rows", () => {
         let view: ShallowWrapper
         beforeAll(() => {
-            const chart = setupGrapher(792, [3512])
-            view = shallow(<DataTable grapher={chart} />)
+            const grapher = new Grapher(
+                {
+                    hasMapTab: true,
+                    tab: "map",
+                    map: {
+                        timeTolerance: 5,
+                    },
+                    dimensions: [
+                        {
+                            variableId: 3512,
+                            property: "y",
+                            display: {
+                                name: "",
+                                unit: "% of children under 5",
+                                tolerance: 5,
+                                isProjection: false,
+                            },
+                        },
+                    ],
+                    owidDataset: {
+                        variables: {
+                            "3512": {
+                                years: [2000, 2010, 2010],
+                                entities: [207, 15, 207],
+                                values: [4, 20, 34],
+                                id: 3512,
+                                shortUnit: "%",
+                            },
+                        },
+                        entityKey: {
+                            "15": { name: "Afghanistan", id: 15, code: "AFG" },
+                            "207": { name: "Iceland", id: 207, code: "ISL" },
+                        },
+                    },
+                },
+                {
+                    queryStr: "?time=2002",
+                }
+            )
+            view = shallow(<DataTable grapher={grapher} />)
         })
 
         it("renders no value when data is not available for years within the tolerance", () => {
-            const cell = view.find("tbody .dimension").at(1).first()
+            const cell = view.find("tbody .dimension").at(0).first()
             expect(cell.text()).toBe("")
         })
 
-        it("renders a tolerance notice when data is not from targetYear", () => {
-            const notice = view
-                .find("tbody tr td.dimension")
-                .first()
-                .find(ClosestYearNotice)
-            expect(notice.prop("closestYear")).toBe("2013")
-            expect(notice.prop("targetYear")).toBe("2016")
-        })
+        // todo: reenable
```

Comment:
@shaahmad I was struggling a bit to figure out tolerance and what combination of setup I needed to get this notice to appear. Any tips?

Reply:
DM-ing